uefi.go

```diff
@@ -46,28 +52,49 @@ func (u UEFIFirmwareDevice) QemuParams(config *Config) []string {
 // template file before using it in a running VM.
 func NewSystemUEFIFirmwareDevice(useSecureBoot bool) (*UEFIFirmwareDevice, error) {
 	uefiDev := UEFIFirmwareDevice{}
-
-	if useSecureBoot {
-		uefiDev.Code = SecCodePath
-		if PathExists(UbuntuSecVars) {
-			uefiDev.Vars = UbuntuSecVars
-		} else if PathExists(CentosSecVars) {
-			uefiDev.Vars = CentosSecVars
-		} else {
-			return &uefiDev, fmt.Errorf("secureboot requested, but no secureboot OVMF variables found")
-		}
-	} else {
-		if PathExists(UnSecCodePath) {
-			uefiDev.Code = UnSecCodePath
-		} else {
+	switch runtime.GOARCH {
+	case "s390x", "amd64":
+		if useSecureBoot {
 			uefiDev.Code = SecCodePath
+			if PathExists(UbuntuSecVars) {
+				uefiDev.Vars = UbuntuSecVars
+			} else if PathExists(CentosSecVars) {
+				uefiDev.Vars = CentosSecVars
+			} else {
+				return &uefiDev, fmt.Errorf("secureboot requested, but no secureboot OVMF variables found")
+			}
+		} else {
+			if PathExists(UnSecCodePath) {
+				uefiDev.Code = UnSecCodePath
+			} else {
+				uefiDev.Code = SecCodePath
+			}
+			if PathExists(UnSecVarsPath) {
+				uefiDev.Vars = UnSecVarsPath
+			} else {
+				return &uefiDev, fmt.Errorf("OMVF variables template missing: %s", UnSecVarsPath)
+			}
 		}
-		if PathExists(UnSecVarsPath) {
-			uefiDev.Vars = UnSecVarsPath
+	case "arm64", "aarch64":
+		if useSecureBoot {
+			uefiDev.Code = SecCodePathArm
+			if PathExists(UbuntuSecVarsArm) {
+				uefiDev.Vars = UbuntuSecVarsArm
+			} else {
+				return &uefiDev, fmt.Errorf("secureboot requested, but no secureboot AAVMF variables found")
+			}
 		} else {
-			return &uefiDev, fmt.Errorf("OMVF variables template missing: %s", UnSecVarsPath)
+			if PathExists(UnSecCodePathArm) {
+				uefiDev.Code = UnSecCodePathArm
+			} else {
+				uefiDev.Code = SecCodePathArm
+			}
+			if PathExists(UnSecVarsPathArm) {
+				uefiDev.Vars = UnSecVarsPathArm
+			} else {
+				return &uefiDev, fmt.Errorf("AAVMF variables template missing: %s", UnSecVarsPathArm)
+			}
```

Comment:
Hrm, I don't want to repeat the logic per arch.

What about this:

Define a list of strings for the various buckets.  We have

1. SecCode
2. SecVars
3. UnSecCode
4. UnSecVars

```
var SecCodePaths = []string{SecCodePath, SecCodePathAarch64}
var UnSecCodePaths = []string{UnSecCodePath, UnSecCodePathAarch64}
```

Then we can have a common logic path, replacing our if/else with a for loop

```
for _, code := range SecCodePaths {
    if PathExists(code) {
        uefiDev.Code = path
        break
    }
}
if uefiDev.Code == "" {
    return &uefiDev, fmt.Errorf("Failed to find UEFI Code firmware")
}
```

you'll have to adapt this pattern to the different if/else clauses but
I think that works better; and we won't need to use the switch after all.

Reply:
Would it be fine to write some helper functions to handle the looping. I think I can reduce some redundancy in the if/else cases that way [ex. when booting without secureboot, we first check to see if the regular firmware blobs are there, then the secure ones]